builtin/aws/lambda/platform.go

```diff
@@ -0,0 +1,706 @@
+package lambda
+
+import (
+	"context"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/subtle"
+	"crypto/x509"
+	"encoding/base64"
+	"fmt"
+	"io"
+	"net"
+	"reflect"
+	"strconv"
+	"strings"
+	"time"
+
+	"golang.org/x/crypto/ssh"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/elbv2"
+	"github.com/aws/aws-sdk-go/service/iam"
+	"github.com/aws/aws-sdk-go/service/lambda"
+	"github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/waypoint/builtin/aws/ecr"
+	"github.com/hashicorp/waypoint/builtin/aws/utils"
+	"github.com/pkg/errors"
+
+	"github.com/hashicorp/waypoint-plugin-sdk/component"
+	"github.com/hashicorp/waypoint-plugin-sdk/docs"
+	"github.com/hashicorp/waypoint-plugin-sdk/terminal"
+)
+
+// Platform is the Platform implementation for AWS Lambda
+type Platform struct {
+	config Config
+}
+
+// ConfigSet is called after a configuration has been decoded
+// we can use this to validate the config
+func (p *Platform) ConfigSet(config interface{}) error {
+	_, ok := config.(*Config)
+	if !ok {
+		// this should never happen
+		return fmt.Errorf("Invalid configuration, expected *lambda.Config, got %s", reflect.TypeOf(config))
+	}
+
+	return nil
+}
+
+// Config implements Configurable
+func (p *Platform) Config() (interface{}, error) {
+	return &p.config, nil
+}
+
+// DeployFunc implements component.Platform
+func (p *Platform) DeployFunc() interface{} {
+	return p.Deploy
+}
+
+// DestroyFunc implements component.Destroyer
+func (p *Platform) DestroyFunc() interface{} {
+	return p.Destroy
```

Comment:
Do we need a `DestroyWorkspace` implementation too to clean up the other stuff you setup in here? or is it all container per-deployment?

Reply:
This part is what prompted my question [in Slack](https://hashicorp.slack.com/archives/C013QT1KG9W/p1612482100254500).
We create a few things as part of the workspace (e.g. Task Role) that we don't cleanup but probably should.

Reply:
Added it!